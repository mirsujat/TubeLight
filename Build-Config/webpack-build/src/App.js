import React, { Component } from 'react';
import { Link, Route } from 'react-router-dom';

import Users from './containers/Users';
import asyncComponent from './hoc/asyncComponent';

const AsyncPizza = asyncComponent(() => {
	return import('./containers/Pizza');
});

class App extends Component {
	render() {
		return (
			<div>
				<div>
					<Link to="/">Users</Link> | <Link to="/pizza">Pizza</Link>
				</div>
				<div>
					<Route path="/" component={Users} />
					<Route path="/pizza" component={AsyncPizza} />
				</div>
			</div>
		);
	}
}

export default App;
