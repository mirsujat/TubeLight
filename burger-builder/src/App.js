import React, { Component } from 'react';


class App extends Component {
  render() {
    return (
      <div>My React App!!</div>
    );
  }
}

export default App;
