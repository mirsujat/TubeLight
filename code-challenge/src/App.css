* {
  box-sizing: border-box;
}
.layout {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0;
  padding: 0;
}
.nav-bar {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  padding: 0 25px;
  margin: 0;
  width: 100%;
  height: 60px;
  background-color: teal;
  color: #f1f1f1;
}
ul {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0;
  margin: 0;
}
li {
  list-style: none;
  padding: 15px;
  margin: 0;
  cursor: pointer;
}
form {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}
input {
  display: block;
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.1);
  line-height: 18px;
  padding: 4px 10px;
}
button {
  line-height: 20px;
  padding: 3px 15px;
  font-size: 16px;
  cursor: pointer;
}
label {
  font-weight: 600;
  color: rgba(0, 0, 0, 0.68);
}
a {
  color: #f1f1f1;
  text-decoration: none;
}
.app {
  display: block;
  position: relative;
  min-height: 1px;
  min-width: 1px;
  padding: 0;
  margin: 0;
}
.nav-right {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
}

.main {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0;
  margin-top: 40px;
  margin-right: 0;
  margin-bottom: 40px;
  margin-left: 0;
  min-height: calc(100vh - 120px);
}

.footer {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0 25px;
  margin: 0;
  height: 60px;
  width: 100%;
  bottom: 0;
  background-color: teal;
  color: #f1f1f1;
}
.nav-bottom {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}
.modal-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0;
  margin: 0;
}
.modal {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-width: 500px;
  height: auto;
  padding: 25px 40px;
  margin: 0;
  position: absolute;
  top: 11%;
  z-index: 10000;
  background-color: white;
  color: rgba(0, 0, 0, 0.84);
  box-shadow: 10px 14px 20px 9px rgba(0, 0, 0, 0.2);
}
.backdrop {
  display: block;
  position: absolute;
  width: 100%;
  height: auto;
  top: 0;
  right: 0;
  bottom: -895px;
  left: 0;
  z-index: 100;
  background-color: rgba(0, 0, 0, 0.4);
  cursor: pointer;
}
.form-group {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}
.reg-form {
  width: 100%;
}
.form-field {
  display: block;
  width: 100%;
  padding: 8px 0;
  margin: 0;
}

.flat-btn {
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.1);
  padding: 5px;
  background-color: white;
  color: rgba(0, 0, 0, 0.64);
}
.red {
  color: red;
  font-size: 12px;
}
p {
  padding: 3px 0;
  margin: 0;
}
.submit {
  color: green;
}

.card {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding: 0;
  margin: 8px;
  color: rgba(0, 0, 0, 0.84);
  background-color: white;
  width: 320px;
  height: 250px;
  box-shadow: 3px 1px 15px 6px rgba(0, 0, 0, 0.2);
}
.card h1 {
  text-decoration: underline;
  color: rgba(0, 0, 0, 0.64);
}
.products {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.buy {
  border: 1px solid #f1f1f1;
  background-color: #ff9800;
  color: #f1f1f1;
  padding: 6px 25px;
}

.cart {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 61px;
  right: 12px;
  padding: 25px;
  color: rgba(0, 0, 0, 0.84);
  background-color: white;
  width: auto;
  height: auto;
  box-shadow: 3px 1px 15px 6px rgba(0, 0, 0, 0.2);
}

table,
thead,
td {
  border: 1px solid rgba(0, 0, 0, 0.1);
}
thead {
  background-color: rgba(0, 0, 0, 0.1);
}
thead > tr > td {
  padding: 6px;
  text-align: left;
  font-weight: 600;
  font-size: 16px;
  color: rgba(0, 0, 0, 0.74);
}
tbody > tr > td {
  padding: 5px 15px;
  text-align: left;
  font-weight: 400;
  color: rgba(0, 0, 0, 0.8);
}
tfoot > tr > td {
  padding: 5px 15px;
  text-align: left;
  font-weight: 700;
  color: rgba(0, 0, 0, 0.64);
}

caption {
  display: table-caption;
  text-align: center;
  font-size: 32px;
  font-weight: 700;
  /* color: rgba(0, 0, 0, 0.64); */
  padding-bottom: 15px;
  color: #ff9800;
}
table {
  border-collapse: collapse;
  width: 100%;
}
.empty {
  border: 0px solid white;
}
.order-form {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  width: 650px;
  height: 350px;
}
