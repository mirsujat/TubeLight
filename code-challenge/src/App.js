import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";

import Layout from "./Demo-1/Layout/Layout";
import Home from "./Demo-1/Pages/Home";
import Contact from "./Demo-1/Pages/Contact";
import ProductList from "./Demo-1/Pages/Ecom/Products/Products";

class App extends Component {
  state = {};
  render() {
    let routes = (
      <Switch>
        <Route path="/" exact component={Home} />
        <Route path="/contact" exact component={Contact} />
        <Route path="/products" exact component={ProductList} />
      </Switch>
    );
    return <div className="app">{routes}</div>;
  }
}

export default App;
