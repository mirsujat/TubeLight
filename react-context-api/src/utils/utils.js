import { applyMiddleware, createStore } from "redux";
import { rootReducer } from "../redux/root-reducer";
import { middlewares } from "../redux/store";

export const findByDataAttr = (component, attr) => {
  const wrapper = component.find(`[data-testid='${attr}']`);
  return wrapper;
};

export const testStore = initialState => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );
  return createStoreWithMiddleware(rootReducer, initialState);
};
