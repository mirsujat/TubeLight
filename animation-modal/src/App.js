import React, { Component } from "react";

import "./App.css";
import Modal from "./components/Modal/Modal";
import Backdrop from "./components/Backdrop/Backdrop";
import List from "./components/List/List";

class App extends Component {
  state = {
    modalIsOpen: false
  }

  showModal = () => {
    this.setState({modalIsOpen: true});
  }

  closeModal = () => {
    this.setState({modalIsOpen: false});
  }

  render() {
    return (
      <div className="App">
        <h1>React Animations</h1>
        <Modal show={this.state.modalIsOpen} closed={this.closeModal}/>
        <Backdrop show={this.state.modalIsOpen} />
        <button className="Button" onClick={this.showModal}>Open Modal</button>
        <h3>Animating Lists</h3>
        <List />
      </div>
    );
  }
}

export default App;
