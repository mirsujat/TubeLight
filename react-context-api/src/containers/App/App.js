import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import "./App.css";

import { setCurrentUser } from "../../redux/user/user.actions";
import { selectCurrentUser } from "../../redux/user/user.selectors";
import MainComponent from "../../components/main/main.component";

class App extends React.Component {
  render() {
    return <MainComponent {...this.props}></MainComponent>;
  }
}

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser
});

const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
