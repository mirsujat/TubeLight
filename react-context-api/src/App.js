import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import "./App.css";

import HeaderContainer from "./containers/Header/HeaderContainer";
import HomePage from "./pages/home/home.page";
import ShopPage from "./pages/shop/shop.page";
import SignInAndSignUpPage from "./pages/sign-in-and-sign-up/sign-in-and-sign-up.page";
import CheckoutPage from "./pages/checkout/checkout.page";

import { auth, createUserProfileDocument } from "./firebase/firebase.utils";

import { setCurrentUser } from "./redux/user/user.actions";
import { selectCurrentUser } from "./redux/user/user.selectors";

class App extends React.Component {
  unsubscribeFromAuth = null;

  componentDidMount() {
    const { setCurrentUser } = this.props;

    this.unsubscribeFromAuth = auth.onAuthStateChanged(async userAuth => {
      if (userAuth) {
        const userRef = await createUserProfileDocument(userAuth);

        userRef.onSnapshot(snapShot => {
          setCurrentUser({
            id: snapShot.id,
            ...snapShot.data()
          });
        });
      }

      setCurrentUser(userAuth);
    });
  }

  componentWillUnmount() {
    this.unsubscribeFromAuth();
  }

  render() {
    return (
      <div data-testid="app">
        <HeaderContainer />
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route path="/shop" component={ShopPage} />
          <Route exact path="/checkout" component={CheckoutPage} />
          <Route
            exact
            path="/signin"
            render={() =>
              this.props.currentUser ? (
                <Redirect to="/" />
              ) : (
                <SignInAndSignUpPage />
              )
            }
          />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser
});

const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
