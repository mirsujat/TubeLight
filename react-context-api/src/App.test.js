import React from "react";
import { shallow } from "enzyme";
import { findByDataAttr, testStore } from "./utils/utils";
import App from "./App";

const setUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(<App store={store}></App>)
    .childAt(0)
    .dive();
  // console.log(wrapper.debug());
  return wrapper;
};

describe("Render App Component Without Error", () => {
  let wrapper;
  beforeEach(() => {
    const initialState = {
      user: {},
      cart: [],
      directory: [],
      shop: []
    };
    wrapper = setUp(initialState);
  });
  it("Should render without errors", () => {
    const component = findByDataAttr(wrapper, "app");
    expect(component.length).toBe(1);
  });
  it("Should render App", () => {
    const component = wrapper;
    expect(component).toMatchSnapshot();
  });
});
