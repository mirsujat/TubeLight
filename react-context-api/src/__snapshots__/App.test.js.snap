// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Component Without Error Should render App 1`] = `ShallowWrapper {}`;
